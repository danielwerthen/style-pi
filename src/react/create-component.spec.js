const React = require('react');
const ShallowRenderer = require('react-test-renderer/shallow');
const createComponent = require('./create-component');
const Style = require('../style');

describe('CreateComponent', () => {
  describe('Simple scenario', () => {
    class MyStyle extends Style {
      methodMissing(name) {
        return value => `${name}: ${value}`;
      }
    }

    it('should work', () => {
      const renderer = new ShallowRenderer();
      const Component = createComponent({
        Style: MyStyle,
      })('div');
      renderer.render(
        React.createElement(Component, {
          children: 'Daniel',
          fontSize: '14px',
          fontFamily: 'Comic Sans',
        }),
      );
      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
