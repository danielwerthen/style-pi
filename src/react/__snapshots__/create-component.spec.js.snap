// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateComponent Simple scenario should work 1`] = `
<div
  className="fontSize: 14px fontFamily: Comic Sans"
>
  Daniel
</div>
`;
